src/fmgc/src/guidance/lnav/legs/FA.ts

```diff
@@ -0,0 +1,233 @@
+import { EfisSymbols } from '@fmgc/efis/EfisSymbols';
+import { Coordinates } from '@fmgc/flightplanning/data/geo';
+import { GuidanceParameters } from '@fmgc/guidance/ControlLaws';
+import { courseToFixDistanceToGo, fixToFixGuidance, getAlongTrackDistanceTo } from '@fmgc/guidance/lnav/CommonGeometry';
+import { LegMetadata } from '@fmgc/guidance/lnav/legs';
+import { Leg } from '@fmgc/guidance/lnav/legs/Leg';
+import { PathVector, PathVectorType } from '@fmgc/guidance/lnav/PathVector';
+import { LnavConfig } from '@fmgc/guidance/LnavConfig';
+import { getNavigation } from '@fmgc/navigation/Navigation';
+import { SegmentType } from '@fmgc/wtsdk';
+import { GeometryNdSymbol } from '@shared/NavigationDisplay';
+
+export class FALeg extends Leg {
+    private wasActive = false;
+
+    private termConditionMet = false;
+
+    private startPoint: Coordinates;
+
+    private endPoint: Coordinates;
+
+    private predictedDistance: NauticalMiles = 1;
+
+    constructor(
+        public fix: WayPoint,
+        private course: DegreesTrue,
+        private altitude: Feet,
+        public readonly metadata: Readonly<LegMetadata>,
+        public segment: SegmentType,
+    ) {
+        super();
+
+        this.startPoint = fix.infos.coordinates;
+
+        this.endPoint = Avionics.Utils.bearingDistanceToCoordinates(
+            this.course,
+            this.predictedDistance,
+            this.startPoint.lat,
+            this.startPoint.long,
+        );
+    }
+
+    get inboundCourse(): Degrees {
+        return this.course;
+    }
+
+    get outboundCourse(): Degrees {
+        return this.course;
+    }
+
+    get terminationWaypoint(): Coordinates {
+        return this.endPoint;
+    }
+
+    get ident(): string {
+        return Math.round(this.altitude).toString();
+    }
+
+    get mapSymbols(): GeometryNdSymbol[] {
+        const symbols = [];
+
+        const followingDiscont = !this.inboundGuidable;
+
+        if (followingDiscont) {
+            symbols.push({
+                databaseId: this.fix.icao,
+                ident: this.fix.ident,
+                location: this.fix.infos.coordinates,
+            });
+        }
+
+        symbols.push({
+            databaseId: EfisSymbols.tempDatabaseId(this.ident),
+            ident: this.ident,
+            location: this.getPathEndPoint(),
+            speedConstraint: EfisSymbols.mapSpeedConstraintFromMetadata(this.metadata.speedConstraint),
+        });
+
+        return symbols;
+    }
```

Comment:
I don't like this... I think the discont logic + extra symbol can be figured out in the EfisSymbols logic..